### ISSUE TITLE
not sure to understand multilevel

### ISSUE THREAD

**Issue by ysagon** [SteveyPugs/fixy#17]
Hello,

I'm using your library with single level strings and I'm very happy with it, thanks.

Now I tried to parse this:
   
    Type            : Explicit
    Target          : ViVol: anamedvivol
    Usage           : 1.435 TB
      Limit         : 3 TB (Hard)
      Warning       : 75% (2.25 TB)
      Critical      : 85% (2.55 TB)
      Reset         : 5% (153.6 GB)
    File Count      : 1535142
      Limit         : Unset
      Warning       : 75% (0)
      Critical      : 85% (0)
      Reset         : 5% (0)
    Generate Events : Disabled

But I'm not sure how to proceed. Does this work with what you call multilevel? If yes, do I need to create a level per line? The lines aren't of a fixed length in this case. The order is always the same, so I'm only interested by the values, not the key. Thanks

**Comment by SteveyPugs** [comment]
Hey @ysagon.

The example you are showing above is a type of data that cannot be parsed by this project as it is horizontal column based not vertical. This open source software handles fixed width / length strings (or files).

Put simply lets say you get a file that is something like this example:
```
SteveyPugs   M321985
Ysagon       M311986
Test         F251992
```
No normal system can parse the example as spaces are different size amounts as well as the string amounts that are in each columns. One thing is consistent which is the point of where text will stop and start.

The first set of data here covers a name, the first 13 chars of each line.
The second covers the sex of the person which is the next 1 chars,
The third covers age which is 2 more chars.
Last the fourth is the birth year which covers 4 more chars.

In total you have 20 chars per line but without a definitive separation value, you would never know where column 1 ends and column 2 starts. In order to do a parse you need to have the map, which I listed above.

Let me know if you have any other questions!

**Comment by ysagon** [comment]
Hello,

thanks for the explanation, but as said, I'm using already your library for an other vertical based data set without problem. I wasn't just not sure about the multilevel stuff (I'm still not sure what is the purpose of this mode). So I guess I'll do the parsing manually for this case.

Best

**Comment by SteveyPugs** [comment]
My apologies. Multilevel is for files that are not the same length on each row. For example:

'''
20170529 Date
SteveyPugs   M321985
Ysagon       M311986
Test         F251992
022800 Time
'''

In this file the first row, last row and 2-4 are completely different setups. You need multiple mapping schemes to parse it. Multi-Level helps with this.

**Comment by ysagon** [comment]
Ok I see, thanks for the explanation. Strange format anyway:)

**Comment by SteveyPugs** [comment]
Can you clarify what you mean by strange format? Do you mean the file or strings being passed in for parsing?

**Comment by ysagon** [comment]
Yes the example input string. But I guess some software may output such hard to parse strings.

**Comment by SteveyPugs** [comment]
Ah. Yes it definitely is. You can parse it in excel as well.
